pkgs/applications/networking/soulseek/nicotine-plus/default.nix

```
@@ -1,44 +1,116 @@
-{ stdenv, fetchFromGitHub, python27Packages, geoip }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3
+, wrapGAppsHook
+, gobject-introspection
+, gtk3
+, libnotify
+, geoip
 
-with stdenv.lib;
+# Test dependencies
+, xvfb_run
+, gnome3
+}:
 
-python27Packages.buildPythonApplication {
+python3.pkgs.buildPythonApplication {
   pname = "nicotine-plus";
-  version = "1.4.1";
+  version = "unstable-2020-05-24";
 
   src = fetchFromGitHub {
     owner = "Nicotine-Plus";
     repo = "nicotine-plus";
-    rev = "4e057d64184885c63488d4213ade3233bd33e67b";
-    sha256 = "11j2qm67sszfqq730czsr2zmpgkghsb50556ax1vlpm7rw3gm33c";
+    rev = "0f9e4e1c2391196d4070df5851ff48601aceaf7f";
+    hash = "sha256-kSDqoxmX9beqL98tMQylbhCZVMmQy6/oVOyW1xnxFLY=";
   };
 
-  propagatedBuildInputs = with python27Packages; [
-    pygtk
+  nativeBuildInputs = [ wrapGAppsHook ];
+
+  # Avoid double-wrapping; see preFixup
+  # and https://nixos.org/nixpkgs/manual/#ssec-gnome-common-issues.
+  dontWrapGApps = true;
+
+  buildInputs = [
+    gobject-introspection
+    gtk3
+    libnotify
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pygobject3
     miniupnpc
     mutagen
-    notify
     (GeoIP.override { inherit geoip; })
   ];
 
-  # Insert real docs directory.
-  # os.getcwd() is not needed
+  checkInputs = [ python3.pkgs.pytest ] ++ lib.optionals (!stdenv.isDarwin) [
+    # The gtk3 package doesn't enable X11 on Darwin, so we can't
+    # use xvfb-run.
+    xvfb_run
+    python3.pkgs.robotframework
+    # Avoid warnings about missing icons.
+    gnome3.defaultIconTheme
+  ];
+
+  # Run setup hooks for installCheckPhase;
+  # see https://github.com/NixOS/nixpkgs/issues/56943.
+  strictDeps = false;
+
   postPatch = ''
-    substituteInPlace ./pynicotine/gtkgui/frame.py \
-      --replace "paths.append(os.getcwd())" "paths.append('"$out"/doc')"
+    # Remove non-free files.
+    sh debian/nicotine-rm-nonfree
```

```suggestion
    sh debian/nicotine-rm-nonfree
```

Please reference bash from nixpkgs.